src/s1ap/handlers/s1ap_msg_delegator.c

```diff
@@ -80,6 +80,11 @@ int convertToInitUeProtoIe(InitiatingMessage_t *msg, struct proto_IE* proto_ies,
 							return -1;
 						}
 
+                        if (s1apNASPDU_p->size > 300) {
```

Comment:
How did you arrive at 300?

Reply:
The destination of the `memcpy` is `s1Msg->nasMsg.nasMsgBuf`. `s1Msg` is of type `initial_ue_msg_t`, which has the following definition:

https://github.com/omec-project/Nucleus/blob/a8002eb10e10554593ebd5a94ec2d3fdb8ee467a/include/common/msgType.h#L603-L614

Following from that, the `rawNas_Q_msg_t` type has definition:

https://github.com/omec-project/Nucleus/blob/a8002eb10e10554593ebd5a94ec2d3fdb8ee467a/include/common/msgType.h#L114-L118

... which actually uses MAX_NAS_MSG_SIZE, equal to 500 bytes. My bad. I think I had seen the general `nasMsgBuf` field in several other types having a constant of 300, such as:

https://github.com/omec-project/Nucleus/blob/a8002eb10e10554593ebd5a94ec2d3fdb8ee467a/include/common/msgType.h#L339-L347

... and wrongly assumed they would be consistent across all message types. I'll get that fixed.

Reply:
The destination of the `memcpy` is `s1Msg->nasMsg.nasMsgBuf`. `s1Msg` is of type `initial_ue_msg_t`, which has the following definition:

https://github.com/omec-project/Nucleus/blob/a8002eb10e10554593ebd5a94ec2d3fdb8ee467a/include/common/msgType.h#L603-L614

Following from that, the `rawNas_Q_msg_t` type has definition:

https://github.com/omec-project/Nucleus/blob/a8002eb10e10554593ebd5a94ec2d3fdb8ee467a/include/common/msgType.h#L114-L118

... which actually uses MAX_NAS_MSG_SIZE, equal to 500 bytes. My bad. I think I had seen the general `nasMsgBuf` field in several other types having a constant of 300, such as:

https://github.com/omec-project/Nucleus/blob/a8002eb10e10554593ebd5a94ec2d3fdb8ee467a/include/common/msgType.h#L339-L347

... and wrongly assumed they would be consistent across all message types. I'll get that fixed.